File: Task 1.py
def convert_temperature(value, scale_from, scale_to):
    if scale_from == scale_to:
        return value
    if scale_from == "Celsius":
        if scale_to == "Fahrenheit":
            return (value * 9/5) + 32
        elif scale_to == "Kelvin":
            return value + 273.15
    elif scale_from == "Fahrenheit":
        if scale_to == "Celsius":
            return (value - 32) * 5/9
        elif scale_to == "Kelvin":
            return (value - 32) * 5/9 + 273.15
    elif scale_from == "Kelvin":
        if scale_to == "Celsius":
            return value - 273.15
        elif scale_to == "Fahrenheit":
            return (value - 273.15) * 9/5 + 32


temp = float(input("Enter the temperature value: "))
from_scale = input("From (Celsius/Fahrenheit/Kelvin): ")
to_scale = input("To (Celsius/Fahrenheit/Kelvin): ")
result = convert_temperature(temp, from_scale, to_scale)
print(f"{temp}° {from_scale} = {result:.2f}° {to_scale}")

File: Task 1.java
import java.util.Scanner;

public class TemperatureConverter {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter temperature value: ");
        double temp = sc.nextDouble();

        System.out.print("From (Celsius/Fahrenheit/Kelvin): ");
        String from = sc.next();

        System.out.print("To (Celsius/Fahrenheit/Kelvin): ");
        String to = sc.next();

        double result = convert(temp, from.toLowerCase(), to.toLowerCase());
        System.out.printf("%.2f° %s = %.2f° %s\n", temp, from, result, to);
    }

    static double convert(double value, String from, String to) {
        if (from.equals(to)) return value;

        if (from.equals("celsius")) {
            if (to.equals("fahrenheit")) return (value * 9 / 5) + 32;
            else if (to.equals("kelvin")) return value + 273.15;
        } else if (from.equals("fahrenheit")) {
            if (to.equals("celsius")) return (value - 32) * 5 / 9;
            else if (to.equals("kelvin")) return (value - 32) * 5 / 9 + 273.15;
        } else if (from.equals("kelvin")) {
            if (to.equals("celsius")) return value - 273.15;
            else if (to.equals("fahrenheit")) return (value - 273.15) * 9 / 5 + 32;
        }
        return 0;
    }
}
